src/domain/mongoose_domain_core.erl

```diff
@@ -0,0 +1,139 @@
+%% Generally, you should not call anything from this module.
```

Comment:
This module doesn't correspond to initial requirements:
* ETS must be read only for public access
* init must accept 2 parameters:
    * the list of initial (not changeable) host_type/domain pairs
    * the list of acceptable host_type
* host_type cannot be changed for the existing domain, it should be possible only to add or remove domain.
* domains can be added only for acceptable host_types

Reply:
> ETS must be read only for public access

Moving logic inside a core gen_server could crash the server, which would get ETS table lost. Less logic - less ways to fail. It also would make debugging and testing harder.

Reply:
This solution is not acceptable, there won't be too much code for gen_server and that is very easy to test.
Later on we will have to introduce hooks for adding and removing domains (executed on each and every node). Current implementation doesn't allow to do this.

Reply:
ok, fine. I will make it slower and with more code.

Reply:
as for the hooks, we may want to do execution asynchronously to prevent blocking operations on the ets. We can design it later (if required at all), for now we just need to ensure that there's an option to implement it.